packages/codegen/src/babel-plugin.ts

```diff
@@ -0,0 +1,94 @@
+import type { PluginObj } from '@babel/core';
+import { template } from '@babel/core';
+import { declare } from '@babel/helper-plugin-utils';
+import type { NodePath } from '@babel/traverse';
+import type { Program } from '@babel/types';
+import * as nodePath from 'node:path';
+import * as t from '@babel/types';
+import { changeExtension } from './utils.js';
+
+interface PluginOptions {
+  tagName: string;
+}
+
+function createImportName(name: string): string {
+  if (name.endsWith('Fragment')) {
+    return `${name}Doc`;
+  } else if (name.endsWith('Query')) {
+    return name.replace('Query', 'Document');
+  } else {
+    return name.replace('Mutation', 'Document');
+  }
+}
+
+const babelPlugin = declare<PluginOptions>((api, options): PluginObj => {
+  let program: NodePath<Program>;
+
+  return {
+    name: 'data-eden-codegen',
+    visitor: {
+      Program(path) {
+        program = path;
+      },
+      TaggedTemplateExpression(path, state) {
+        const { node } = path;
+
+        if (!(t.isIdentifier(node.tag) && node.tag.name === options.tagName)) {
```

Comment:
This is not good (reserving a whole variable name). We should be matching an import and allow aliasing. See https://github.com/emberjs/babel-plugin-ember-template-compilation/blob/main/src/plugin.ts#L143 for an example of what we do for `hbs` in Ember land.

Reply:
https://github.com/data-eden/data-eden/issues/99